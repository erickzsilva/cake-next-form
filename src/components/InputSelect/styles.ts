import styled from 'styled-components'

export const Select = styled.select`
  border: 1px solid #eba70e;
  border-radius: 3px;
  box-shadow: none;
  color: #24457b;
  font-size: 11px;
  height: 35px;
  padding-left: 5px;
`
