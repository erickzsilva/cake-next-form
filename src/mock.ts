import { CakeResponse } from './sharedTypes/types'

export const mock: CakeResponse = {
  title: 'Cake Order Form',
  description: `Order your freshly baked cakes using only the finest quality natural ingredientes.`,
  textChooseYourCake:
    'Please choose your favorite cake from among the following',
  subTitle: 'Order Information',
  cakes: [
    {
      value: 'Red Fruit Cake',
      name: 'redFruitCake',
      alt: 'Red Fruit Cake',
      srcImage: 'img/redFruitCake.jpg'
    },
    {
      value: 'Strawberry Sponge Cake',
      name: 'strawberrySpongeCake',
      alt: 'Strawberry Sponge Cake',
      srcImage: 'img/strawberrySpongeCake.jpg'
    },
    {
      value: 'Hazelnut Pie',
      name: 'hazelnutPie',
      alt: 'Hazelnut Pie',
      srcImage: 'img/hazelnutPie.jpg'
    },
    {
      value: 'MM Cake',
      name: 'mmCake',
      alt: 'MM Cake',
      srcImage: 'img/mmCake.jpg'
    }
  ],
  country: ['Argentina', 'Brasil', 'Fran√ßa', 'Mexico', 'Paraguai', 'Portugal']
}

export const mockHead = {
  title: 'Cake Order Form',
  description: `Order your freshly baked cakes using only the finest quality natural ingredientes.`
}
