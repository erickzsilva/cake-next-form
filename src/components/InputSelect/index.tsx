import { UseFormRegisterReturn } from 'react-hook-form'
import * as S from './styles'

export type SelectProps = {
  autoComplete?: string
  id: string
  optionList: string[]
  placeHolder: string
  register?: UseFormRegisterReturn
}

const InputSelect = ({
  autoComplete,
  id,
  optionList,
  placeHolder,
  register
}: SelectProps) => (
  <S.Select
    id={id}
    {...register}
    autoComplete={autoComplete}
    data-testid="select"
  >
    <option value="" defaultChecked hidden>
      {placeHolder}
    </option>
    {optionList.map((item: string, index) => (
      <option key={index} value={item}>
        {item}
      </option>
    ))}
  </S.Select>
)

export default InputSelect
