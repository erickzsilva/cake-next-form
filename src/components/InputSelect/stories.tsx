import { Meta } from '@storybook/react'
import InputSelect, { SelectProps } from '.'
import { FormInputs } from '@/sharedTypes/types'
import { useForm } from 'react-hook-form'
import { mock } from './../../mock'

export default {
  title: 'InputSelect',
  component: InputSelect,
  args: {
    id: 'country',
    optionList: mock.country,
    placeHolder: 'Country',
    autoComplete: 'country-name'
  }
} as Meta<SelectProps>

export const Default = (args: SelectProps) => {
  const { register } = useForm<FormInputs>()
  return <InputSelect {...args} register={register('country')} />
}
