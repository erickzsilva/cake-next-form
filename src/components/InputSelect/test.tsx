import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import InputSelect from '.'
import { mock } from '../../mock'

describe('<InputSelect />', () => {
  it('should be rendered and selected an option', () => {
    const { container } = render(
      <InputSelect
        id="country"
        optionList={mock.country}
        placeHolder="Country"
        autoComplete="country-name"
      />
    )

    expect(screen.getByTestId('select')).toBeInTheDocument()

    expect(userEvent.selectOptions(screen.getByTestId('select'), 'Brasil'))
      .toBeCalled

    expect(container.firstChild).toMatchSnapshot()
  })
})
