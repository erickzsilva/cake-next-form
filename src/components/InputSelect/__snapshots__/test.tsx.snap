// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputSelect /> should be rendered and selected an option 1`] = `
.c0 {
  border: 1px solid #eba70e;
  border-radius: 3px;
  box-shadow: none;
  color: #24457b;
  font-size: 11px;
  height: 35px;
  padding-left: 5px;
}

<select
  autocomplete="country-name"
  class="c0"
  data-testid="select"
  id="country"
>
  <option
    hidden=""
    value=""
  >
    Country
  </option>
  <option
    value="Argentina"
  >
    Argentina
  </option>
  <option
    value="Brasil"
  >
    Brasil
  </option>
  <option
    value="França"
  >
    França
  </option>
  <option
    value="Mexico"
  >
    Mexico
  </option>
  <option
    value="Paraguai"
  >
    Paraguai
  </option>
  <option
    value="Portugal"
  >
    Portugal
  </option>
</select>
`;
